File: OLSource1/v1/VS_officedev/codesnippet/CSharp/ribbon-object-model-overview_2.cs
        protected override Microsoft.Office.Core.IRibbonExtensibility
            CreateRibbonExtensibilityObject()
        {
            bool myCondition = false;
            if (myCondition == true)
            {
                Ribbon1 tempRibbon = new Ribbon1();
                tempRibbon.tab1.ControlId.ControlIdType =
                    Microsoft.Office.Tools.Ribbon.RibbonControlIdType.Office;
                tempRibbon.tab1.ControlId.OfficeId = "TabHome";
                return Globals.Factory.GetRibbonFactory().CreateRibbonManager( 
                    new Microsoft.Office.Tools.Ribbon.IRibbonExtension[]
                     { tempRibbon });
            }
            else
            {
                Ribbon2 tempRibbon = new Ribbon2();
                tempRibbon.tab1.ControlId.ControlIdType =
                    Microsoft.Office.Tools.Ribbon.RibbonControlIdType.Office;
                tempRibbon.tab1.ControlId.OfficeId = "TabInsert";
                return Globals.Factory.GetRibbonFactory().CreateRibbonManager(
                                    new Microsoft.Office.Tools.Ribbon.IRibbonExtension[] { tempRibbon });
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/ribbon-object-model-overview_2.vb
    Protected Overrides Function CreateRibbonExtensibilityObject() _
        As Microsoft.Office.Core.IRibbonExtensibility
        Dim myCondition As Boolean = True
        If myCondition = True Then
            Dim tempRibbon As New Ribbon1()
            tempRibbon.Tab1.ControlId.ControlIdType = _
                Microsoft.Office.Tools.Ribbon.RibbonControlIdType.Office
            tempRibbon.Tab1.ControlId.OfficeId = "TabHome"
            Return Globals.Factory.GetRibbonFactory.CreateRibbonManager _
                (New Microsoft.Office.Tools.Ribbon.IRibbonExtension() {tempRibbon})
        Else
            Dim tempRibbon As New Ribbon2()
            tempRibbon.Tab1.ControlId.ControlIdType = _
                Microsoft.Office.Tools.Ribbon.RibbonControlIdType.Office
            tempRibbon.Tab1.ControlId.OfficeId = "TabInsert"
            Return Globals.Factory.GetRibbonFactory.CreateRibbonManager _
                (New Microsoft.Office.Tools.Ribbon.IRibbonExtension() {tempRibbon})
        End If
    End Function